docs/architecture/adr/ADR032-nlb-for-mtls.md

```diff
@@ -0,0 +1,19 @@
+# ADR032: NLB for mTLS
+
+## Status
+
+Accepted
+
+## Context
+
+Verify's [doc-checking](https://github.com/alphagov/doc-checking) serviceis secured in part using mTLS. Currently, our clusters are fronted by ALBs which cannot provide mTLS.
+
+The doc-checking service currently runs an nginx that provides the mTLS functionality from within the cluster. In order for GSP to be able to allow something within the cluster to perform mTLS we must run a load balancer that forwards unaltered TCP packets in addition to, or instead of an ALB.
+
+## Decision
+
+We will optionally create and run an NLB in addition to the current ALB for clusters that have a requirement to terminate their own TLS. This NLB will be available at `nlb.$CLUSTER_DOMAIN`.
+
+## Consequences
+
+In some of our clusters we will be running two public load balancers. This may be confusing or unexpected.
```

Comment:
We should also add something in here about maintenance of TLS certs ourselves (and maybe add something about the AWS Certificate Manager that we're using for the ALB in the "Context" section above) - a la cert-manager (deployed as part of istio).